sim/firesim-lib/src/main/cc/fesvr/firesim_tsi.cc

```diff
@@ -48,6 +58,15 @@ void firesim_tsi_t::load_mem_read(addr_t addr, size_t nbytes, void *dst) {
 
 void firesim_tsi_t::tick() { switch_to_host(); }
 
+void firesim_tsi_t::reset() {
+  is_loaded_in_sw = true;
```

Comment:
I think a comment here would be useful. This loop spins in the fesvr thread until the target has indicated that it is loaded.